noti.go

```diff
@@ -60,54 +62,62 @@ func main() {
 		return
 	}
 
-	runUtility()
-
-	if defs := strings.TrimSpace(os.Getenv(defaultEnv)); defs != "" {
-		*banner = strings.Contains(defs, "banner")
-		*speech = strings.Contains(defs, "speech")
-		*pushbullet = strings.Contains(defs, "pushbullet")
-		*slack = strings.Contains(defs, "slack")
-	} else {
-		var explicitSet bool
-		var val bool
-
-		flag.Visit(func(f *flag.Flag) {
-			if f.Name == "b" || f.Name == "banner" {
-				explicitSet = true
-				// Ignoring error, false on error is fine.
-				val, _ = strconv.ParseBool(f.Value.String())
-			}
-		})
+	wasRunSuccessful := runUtility()
 
-		// If the user explicitly set -banner, then use the value that the user
-		// set, but if no banner flag was set, then the default is true.
-		if explicitSet {
-			*banner = val
+	if *onlyOnError != true || wasRunSuccessful != true {
+		if defs := strings.TrimSpace(os.Getenv(defaultEnv)); defs != "" {
+			*banner = strings.Contains(defs, "banner")
+			*speech = strings.Contains(defs, "speech")
+			*pushbullet = strings.Contains(defs, "pushbullet")
+			*slack = strings.Contains(defs, "slack")
 		} else {
-			*banner = true
+			var explicitSet bool
+			var val bool
+
+			flag.Visit(func(f *flag.Flag) {
+				if f.Name == "b" || f.Name == "banner" {
+					explicitSet = true
+					// Ignoring error, false on error is fine.
+					val, _ = strconv.ParseBool(f.Value.String())
+				}
+			})
+
+			// If the user explicitly set -banner, then use the value that the user
+			// set, but if no banner flag was set, then the default is true.
+			if explicitSet {
+				*banner = val
+			} else {
+				*banner = true
+			}
 		}
-	}
 
-	if *banner {
-		bannerNotify()
-	}
-	if *speech {
-		speechNotify()
-	}
-	if *pushbullet {
-		pushbulletNotify()
-	}
-	if *slack {
-		slackNotify()
+		if *banner {
+			bannerNotify()
+		}
+		if *speech {
+			speechNotify()
+		}
+		if *pushbullet {
+			pushbulletNotify()
+		}
+		if *slack {
+			slackNotify()
+		}
+
+		if wasRunSuccessful != true {
+			os.Exit(1)
+		}
 	}
```

Comment:
This would help to integrate it in build processes, for example if a test run fails make noti exit with a non zero status